projects/swimlane/ngx-datatable/src/lib/components/header/header.component.ts

```diff
@@ -303,7 +304,9 @@ export class DataTableHeaderComponent {
     this._styleByGroup.left = this.calcStylesByGroup('left');
     this._styleByGroup.center = this.calcStylesByGroup('center');
     this._styleByGroup.right = this.calcStylesByGroup('right');
-    this.cd.detectChanges();
+    if (!(this.cd as ViewRef).destroyed) {
```

Comment:
this can also be achieved without type-casing, see
https://github.com/swimlane/ngx-datatable/pull/1792/files#diff-6faf0f1e3361b28329fc920a924fe52dR165-R314